lib/PartitionedProducerImpl.h

```diff
@@ -17,16 +17,16 @@
  * under the License.
  */
 #include "ProducerImpl.h"
-#include "ClientImpl.h"
 #include <vector>
-
 #include <mutex>
 #include <pulsar/MessageRoutingPolicy.h>
 #include <pulsar/TopicMetadata.h>
 #include <lib/TopicName.h>
 
 namespace pulsar {
 
+class ClientImpl;
+
 class PartitionedProducerImpl : public ProducerImplBase,
```

Comment:
Seems we need to implement the `beforeConnectionChange` for the PartitionedProducerImpl.

Reply:
No. Because it doesn't implement `HandlerBase` and has no `connection_` field. The `ClientConnection` instance doesn't hold any `PartitionedProducerImpl` reference